R/timeseriessummary-data.R

```diff
@@ -71,6 +78,100 @@ formatDataTable <- function(inputData){
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Units", value=metadata[['unit']], indent=8, stringsAsFactors = FALSE))
+    if(!is.null(itData) && !isEmptyOrBlank(itData) && nrow(itData) > 0){
+      mIt <- FALSE
+      if(nrow(itData) > 1){
+        changeNote <- TRUE
+        mIt <- TRUE
+      }
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Interpolation", value=ifelse(mIt,  paste(itData[1,][['type']], "*"), itData[1,][['type']]), indent=8, stringsAsFactors = FALSE))
+    }
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Sub-Location", value=metadata[['sublocation']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="UTC Offset", value=metadata[['utcOffset']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Computation", value=metadata[['computation']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Period", value=metadata[['period']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Publish", value=metadata[['publish']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Description", value=metadata[['description']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Comments", value=metadata[['comment']], indent=8, stringsAsFactors = FALSE))
+    if(!is.null(methodData) && !isEmptyOrBlank(methodData) && nrow(methodData) > 0){
+      mMethod <- FALSE
+      if(nrow(methodData) > 1){
+        changeNote <- TRUE
+        mMethod <- TRUE
+      }
+      
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Measurement Method", value=ifelse(mMethod,  paste(methodData[1,][['methodCode']], "*"), methodData[1,][['methodCode']]), indent=8, stringsAsFactors = FALSE))
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Method Start Time", value=methodData[1,][['startTime']], indent=26, stringsAsFactors = FALSE))
+    }
+    if(!is.null(processorData) && !isEmptyOrBlank(processorData) && nrow(processorData) > 0){
+      mProcessor <- FALSE
+      if(nrow(processorData) > 1){
+        changeNote <- TRUE
+        mProcessor <- TRUE
+      }
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Processing Type", value=ifelse(mProcessor, paste(processorData[1,][['processorType']], "*"), processorData[1,][['processorType']]), indent=8, stringsAsFactors = FALSE))
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Period Start Time", value=processorData[1,][['startTime']], indent=26, stringsAsFactors = FALSE))
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Period End Time", value=processorData[1,][['endTime']], indent=26, stringsAsFactors = FALSE))
+    }
+    
+    #Time Series Extended Attributes
+    extAttrs <- metadata[['extendedAttributes']]
+    if(!is.null(extAttrs) && !isEmptyOrBlank(extAttrs)){
+      accessValue <- ifelse(is.null(extAttrs[['ACCESS_LEVEL']]), " ", extAttrs[['ACCESS_LEVEL']])
```

Comment:
rather than `is.null` for these, consider `isEmptyOrBlank` instead. We created that because it's a more robust way to check for missing information in our data.